### PR TITLE
Including admin JWT on outbound platform calls

### PR DESCRIPTION
**Pull request by mathewc**
Making outbound platform auth more flexible, to support new platform components that support JWTs.

### Pull request checklist

* [x] My changes **do not** require documentation changes
    * [ ] Otherwise: Documentation issue linked to PR
* [x] My changes **should not** be added to the release notes for the next release
    * [ ] Otherwise: I've added my notes to `release_notes.md`
* [ ] My changes **do not** need to be backported to a previous version
    * [x] V3 Backport https://github.com/Azure/azure-functions-host/pull/9198
    * [x] V1 Backport https://github.com/Azure/azure-functions-host/pull/9199
* [x] My changes **do not** require diagnostic events changes
    * Otherwise: I have added/updated all related diagnostic events and their documentation (Documentation issue linked to PR)
* [x] I have added all required tests (Unit tests, E2E tests)

### CODE REVIEW

File: src/WebJobs.Script.WebHost/Security/JwtTokenHelper.cs [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+using Microsoft.Azure.WebJobs.Script.Config;
+using Microsoft.IdentityModel.Tokens;
+
+namespace Microsoft.Azure.WebJobs.Script.WebHost.Security
+{
+    internal static class JwtTokenHelper
+    {
+        public static string CreateToken(DateTime validUntil, string audience = null, string issuer = null, byte[] key = null)
+        {
+            var tokenHandler = new JwtSecurityTokenHandler();
```
Comment (jviau) [inline src/WebJobs.Script.WebHost/Security/JwtTokenHelper.cs:15]: Can we store this as a `readonly static`?

File: src/WebJobs.Script/ScriptConstants.cs [hunk 2]
```diff
@@ -133,6 +133,7 @@ public static class ScriptConstants
         public const string AdminJwtScmValidIssuerFormat = "https://{0}.scm.azurewebsites.net";
         public const string AdminJwtSiteValidIssuerFormat = "https://{0}.azurewebsites.net";
         public const string AdminJwtAppServiceIssuer = "https://appservice.core.azurewebsites.net";
+        public const string AdminJwtAppServiceAudience = "https://appservice.core.azurewebsites.net";
```
Comment (jviau) [inline src/WebJobs.Script/ScriptConstants.cs]: Can we make it clearer in this file when/where the different issuer/audiences are to be used. For example, it is not clear if `AdminJwtAppServiceAudience` is meant to be an incoming audience or outgoing? Some suggestions:

1. Name these without `Issuer` or `Audience` at all, just something like `AppServiceCoreUri`
or...
2. Move the constant closer to the usage
or...
3. Make `Outgoing` or `Incoming` clear either by comment, sub-constant class, or constant name.

**Review by jviau — APPROVED** [review]

File: src/WebJobs.Script.WebHost/Security/JwtTokenHelper.cs (hunk 1 above)
Comment (jviau) [reply src/WebJobs.Script.WebHost/Security/JwtTokenHelper.cs:15]: Disregard this. I see it is a lift & shift refactor, also this class does not guarantee thread-safety. (https://learn.microsoft.com/en-us/previous-versions/visualstudio/dn464181(v=vs.114)?redirectedfrom=MSDN#thread-safety)

File: src/WebJobs.Script/ScriptConstants.cs (hunk 2 above)
Comment (mathewc) [reply src/WebJobs.Script/ScriptConstants.cs]: Done

**Pull request merged**